influxdb/_dataframe_client.py

```diff
@@ -140,13 +203,142 @@ def _convert_dataframe_to_json(self, dataframe, measurement, tags=None,
+                            PeriodIndex.')
+
+        # Create a Series of columns for easier indexing
+        column_series = pd.Series(dataframe.columns)
+
+        if field_columns is None:
+            field_columns = []
+        if tag_columns is None:
+            tag_columns = []
+
+        # Make sure field_columns and tag_columns are lists
+        field_columns = list(field_columns) if list(field_columns) else []
+        tag_columns = list(tag_columns) if list(tag_columns) else []
+
+        # If field columns but no tag columns, assume rest of columns are tags
+        if field_columns and (not tag_columns):
+            tag_columns = list(column_series[~column_series.isin(
+                field_columns)])
+
+        # If no field columns, assume non-tag columns are fields
+        if not field_columns:
+            field_columns = list(column_series[~column_series.isin(
+                tag_columns)])
+
+        precision_factor = {
+            "n": 1,
+            "u": 1e3,
+            "ms": 1e6,
+            "s": 1e9,
+            "m": 1e9 * 60,
+            "h": 1e9 * 3600,
+        }.get(time_precision, 1)
+
+        # Make array of timestamp ints
+        time = ((dataframe.index.to_datetime().values.astype(int) /
+                 precision_factor).astype(int).astype(str))
+
+        # If tag columns exist, make an array of formatted tag keys and values
+        if tag_columns:
+            tag_df = dataframe[tag_columns]
+            tag_df = self._stringify_dataframe(
+                tag_df, numeric_precision, datatype='tag')
+            tags = (',' + (
+                (tag_df.columns.values + '=').tolist() + tag_df)).sum(axis=1)
+            del tag_df
+
+        else:
+            tags = ''
+
+        # Make an array of formatted field keys and values
+        field_df = dataframe[field_columns]
+        field_df = self._stringify_dataframe(
+            field_df, numeric_precision, datatype='field')
+        field_df = (field_df.columns.values + '=').tolist() + field_df
+        field_df[field_df.columns[1:]] = ',' + field_df[field_df.columns[1:]]
+        fields = field_df.sum(axis=1)
+        del field_df
+
+        # Add any global tags to formatted tag strings
+        if global_tags:
+            global_tags = ','.join(['='.join([tag, global_tags[tag]])
+                                    for tag in global_tags])
+            if tag_columns:
+                tags = tags + ',' + global_tags
+            else:
+                tags = ',' + global_tags
+
+        # Generate line protocol string
+        points = (measurement + tags + ' ' + fields + ' ' + time).tolist()
+        return points
+
+    def _stringify_dataframe(self,
+                             dataframe,
+                             numeric_precision,
+                             datatype='field'):
+
+        # Find int and string columns for field-type data
+        int_columns = dataframe.select_dtypes(include=['int']).columns
```

Comment:
Can we change this to `include=['integer']` so other integer subtypes won't be treated as floats?